server/src/test/java/org/elasticsearch/cluster/metadata/MetadataDeleteIndexServiceTests.java

```diff
@@ -129,6 +132,29 @@ public void testDeleteBackingIndexForDataStream() {
             DataStream.getDefaultBackingIndexName(dataStreamName, numIndexToDelete)), IsNull.nullValue());
     }
 
+    public void testDeleteMultipleBackingIndexForDataStream() {
+        int numBackingIndices = randomIntBetween(3, 5);
+        int numBackingIndicesToDelete = randomIntBetween(2, numBackingIndices - 1);
+        String dataStreamName = randomAlphaOfLength(6).toLowerCase(Locale.ROOT);
+        ClusterState before = DataStreamTestHelper.getClusterStateWithDataStreams(
+            List.of(new Tuple<>(dataStreamName, numBackingIndices)), List.of());
+
+        List<Integer> indexNumbersToDelete =
+            randomSubsetOf(numBackingIndicesToDelete, IntStream.rangeClosed(1, numBackingIndices).boxed().collect(Collectors.toList()));
```

Comment:
I think `numBackingIndices` needs to be changed into `numBackingIndices - 1`, otherwise the test tries to delete the write index which would fail.